### PR TITLE
8290180: Convert com/sun/management/UnixOperatingSystemMXBean/GetOpenFileDescriptorCount.sh to jtreg java version

### PR DESCRIPTION
**Pull request by bwhuang-us**
This is a task to convert shell tests to Java tests. The shell scripts to be removed are responsible for compiling Java files and running the tests on Linux systems. This functionality can be achieved with JTReg tag @requires (os.family=="linux").

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8290180](https://bugs.openjdk.org/browse/JDK-8290180): Convert com/sun/management/UnixOperatingSystemMXBean/GetOpenFileDescriptorCount.sh to jtreg java version
 * [JDK-8290181](https://bugs.openjdk.org/browse/JDK-8290181): convert com/sun/management/UnixOperatingSystemMXBean/GetMaxFileDescriptorCount.sh to java version

### Reviewers
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)
 * [Alex Menkov](https://openjdk.org/census#amenkov) (@alexmenkov - **Reviewer**) ⚠️ Review applies to [13d99c69](https://git.openjdk.org/jdk/pull/9503/files/13d99c69ba9bfd51ac33df81fe1d1dc93b22944c)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9503/head:pull/9503` \
`$ git checkout pull/9503`

Update a local copy of the PR: \
`$ git checkout pull/9503` \
`$ git pull https://git.openjdk.org/jdk pull/9503/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9503`

View PR using the GUI difftool: \
`$ git pr show -t 9503`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9503.diff">https://git.openjdk.org/jdk/pull/9503.diff</a>

</details>

### CODE REVIEW

File: test/jdk/com/sun/management/UnixOperatingSystemMXBean/GetMaxFileDescriptorCount.java [hunk 1]
```diff
@@ -22,10 +22,13 @@
  */
 
 /*
- *
+ * @test
  * @bug     4858522
  * @summary Basic unit test of UnixOperatingSystemMXBean.getMaxFileDescriptorCount()
  * @author  Steve Bohne
+ * @requires (os.family=="linux")
```
Comment (lmesnik) [inline test/jdk/com/sun/management/UnixOperatingSystemMXBean/GetMaxFileDescriptorCount.java:29]: please add spaces, braces are not required, see examples in jtreg tests
@requires os.family == "linux"

File: test/jdk/com/sun/management/UnixOperatingSystemMXBean/GetMaxFileDescriptorCount.java [hunk 2]
```diff
@@ -22,10 +22,13 @@
...
+ *
+ * @run main/othervm GetMaxFileDescriptorCount
```
Comment (lmesnik) [inline test/jdk/com/sun/management/UnixOperatingSystemMXBean/GetMaxFileDescriptorCount.java:31]: is othervm required here?

Comment (bwhuang-us) [reply test/jdk/com/sun/management/UnixOperatingSystemMXBean/GetMaxFileDescriptorCount.java:31]: No, it is not needed. I will remove it.

**Review by lmesnik — APPROVED** [review]

**Comment by lmesnik** [comment]
There are some possible cleanups in the tests (comments, unused args, variable name style) but seem out of scope for this PR.

**Comment by bwhuang-us** [comment]
It is not out of scope. I can do the clean up along with this PR. Can you please give me some hints what could be optimized?

**Review by alexmenkov — APPROVED** [review]

**Comment by lmesnik** [comment]
The variables should use camelCase. Like min_count_for_pass -> minCountForPass 
Variable trace not needed, just print count always
The mention of Solaris should be removed from comments, also I don't know if instructions for manual testing are needed, but it would be better to say "to know actual value run smth".

**Comment by dholmes-ora** [comment]
Is the restriction to run on Linux actually still valid? Shouldn't this work on any of our Posix-like  platforms? I suspect you need to read "Linux" as "not Solaris or Windows" when looking at this test (and possibly others).

**Comment by bwhuang-us** [comment]
@lmesnik I've completed removed the comments for manual tests because I don't think manual tests are necessary in this scenario. If the purpose of the tests is to compare FD count from the Java API to the system properties, it can be done in automated tests. Potential test cases could be opening a new file and observing the opened FD count for increment, or checking the value from the system call. We can address this in a separate ticket if we want.

**Comment by alexmenkov** [comment]
Still good to me

**Comment by dholmes-ora** [comment]
> The other question is should we include Solaris as it is also unix based.

@bwhuang-us there is no Solaris support in mainline.

**Pull request closed without merging**